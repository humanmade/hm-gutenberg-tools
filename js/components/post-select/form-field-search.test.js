// import jest from 'jest';
import React from 'react';
import TestRenderer from 'react-test-renderer';

import FormFieldSearch from './form-field-search';

test( 'Form Search Field', () => {
	const onChangeHandler = jest.fn();

	const testRenderer = TestRenderer.create(
		<FormFieldSearch fieldId="id-123" label="Search Field" value="test value" onChange={ onChangeHandler } />
	);

	expect( testRenderer.toJSON() ).toMatchSnapshot();

	testRenderer.root.findByType( 'input' ).props.onChange( { target: { value: 'new value' } } );
	expect( onChangeHandler.mock.calls.length ).toBe( 1 );
	expect( onChangeHandler.mock.calls[0][0] ).toBe( 'new value' );
} );
