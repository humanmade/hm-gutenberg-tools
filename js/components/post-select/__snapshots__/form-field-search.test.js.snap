// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Row 1`] = `
<div
  className="post-select-form-row"
>
  <label
    className="screen-reader-text"
    htmlFor="id-123"
  >
    Search Field
  </label>
  <input
    id="id-123"
    onChange={[Function]}
    placeholder={undefined}
    type="search"
    value="test value"
  />
</div>
`;
